engines/bops_core/app/components/bops_core/table_of_contents_component.rb

```diff
@@ -2,8 +2,41 @@
 
 class BopsCore::TableOfContentsComponent < ViewComponent::Base
   def initialize(items:)
-    @items = items
+    @items = build_items(items)
   end
 
   attr_reader :items
+
+  private
+
+  def build_items(items)
+    Array(items).filter_map do |section|
+      links = visible_links(section[:links])
+      next unless links.any?
+
+      section_hash = section.respond_to?(:to_h) ? section.to_h : section
+      next unless section_hash.respond_to?(:merge)
+
+      section_hash.merge(links: links)
+    end
+  end
+
+  def visible_links(links)
+    Array(links).filter_map do |link|
+      text, href, condition = link
+      next unless link_visible?(condition)
+
+      [text, href]
+    end
```

Comment:
Maybe this can be simpler?
```suggestion
    Array(links).filter_map do |text, href, condition|
      [text, href] if link_visible?(condition)
    end
```

Reply:
Also, both here and on L13 you're using `Array(links)` — but will these methods ever receive something other than an array? It won't hurt but it might not be necessary.